src/v2i-hub/ImmediateForwardPlugin/src/IMFNTCIP1218Worker.cpp

```diff
@@ -0,0 +1,117 @@
+#include "IMFNTCIP1218Worker.h"
+
+using namespace tmx::utils;
+
+namespace ImmediateForward {
+    void clearImmediateForwardTable( tmx::utils::snmp_client* const client) {
+
+        FILE_LOG(logDEBUG) << "Retrieving max Imf rows ..." ;
+        snmp_response_obj maxImfsRep;
+        maxImfsRep.type = snmp_response_obj::response_type::INTEGER;
+        bool connected = client->process_snmp_request(rsu::mib::ntcip1218::maxRsuIFMs, request_type::GET, maxImfsRep);
+        auto maxImfs = maxImfsRep.val_int;
+        FILE_LOG(logDEBUG) << "Max Imf rows " << maxImfs ;
+        auto curIndex = 1;
+        if (connected) {
+            while ( maxImfs >= curIndex && 16 >= curIndex) {
+                snmp_response_obj deleteRowRep;
+                deleteRowRep.type = snmp_response_obj::response_type::INTEGER;
+                deleteRowRep.val_int = 6;
+                std::string oid = rsu::mib::ntcip1218::rsuIFMStatusOid +  "." + std::to_string(curIndex);
+                connected = client->process_snmp_request(oid, request_type::SET, deleteRowRep);
+                if (!connected) {
+                    FILE_LOG(logWARNING) << "Cleared IMF Table Rows 0 to "<< curIndex ;
+                    break;
+                }
+                curIndex++;
+
+            }
+        }
+        
+    }
+
+    void setRSUMode(tmx::utils::snmp_client* const client, unsigned int mode) {
+        snmp_response_obj obj;
+        obj.type = snmp_response_obj::response_type::INTEGER;
+        obj.val_int = mode;
+        bool operational = client->process_snmp_request(rsu::mib::ntcip1218::rsuModeOid, request_type::SET, obj);
+        if (!operational) {
+            throw tmx::TmxException("Failed to set RSU to operational mode");
+        }
+    }
+
+    std::unordered_map<std::string, unsigned int> initializeImmediateForwardTable( snmp_client* const client, const std::vector<Message> &messages){
+        std::unordered_map<std::string, unsigned int> tmxMessageTypeToIMFTableIndex;
+        // Immediate Forward Messages Table index starts with 1
+        auto curIndex = 1;
+        FILE_LOG(logDEBUG1) << "Initializing RSU IMF Table" ;
+        for (const auto &message : messages)
+        {
+            //create new row entry
+            FILE_LOG(logDEBUG1) << "Creating IMF row " + std::to_string(curIndex) ;
+            std::vector<snmpRequest> requests;
+           
+            size_t pos = message.psid.find("x");
+            if (pos == std::string::npos) {
+                throw tmx::TmxException("Message PSID " + message.psid + " is malformed and should be formated 0x<PSID HEX>");
+            }
+            std::string messagePsidwithoutPrefix = message.psid.substr(pos+1);
+            snmpRequest psid{
+                rsu::mib::ntcip1218::rsuIFMPsidOid + "." + std::to_string(curIndex),
+                'x',
+                messagePsidwithoutPrefix
+            };
+            snmpRequest channel{
```

Comment:
The channel request variable is a bit confising. Same comments for below requests, Maybe append with req or request?

Reply:
I am not sure this change makes sense to me. The object class is in the declaration as well as the function call.